blockstore/src/main/java/brooklyn/location/blockstore/BlockDeviceOptions.java

```diff
@@ -27,7 +27,11 @@ public static BlockDeviceOptions fromMap(Map<String, ?> map) {
             }
         }
         if (map.containsKey("sizeInGb")) {
-            result.sizeInGb = (Integer) checkNotNull(map.get("sizeInGb"), "sizeInGb");
+            if (map.get("sizeInGb") instanceof Double && (Double) map.get("sizeInGb") % 1 != 0) {
```

Comment:
Comparing double to 0 with `==` could lead to surprising results. A safer alternative would be: `Math.abs(sizeInGb - (Integer)sizeInGb) < eps` where `eps` is small enough - in our case anything <=  `0.01` should work.

Perhaps include the `Float` case as well? (by casting it to double and doing the same check).